core/src/main/java/hudson/model/UpdateCenter.java

```diff
@@ -969,6 +969,10 @@ private XmlFile getConfigFile() {
         Map<String,Plugin> pluginMap = new LinkedHashMap<>();
         for (UpdateSite site : sites) {
             for (Plugin plugin: site.getAvailables()) {
```

Comment:
The plugin should only be hidden in the update center, not from the update sites as I suspect that will have side effects.